postgresql-plugin/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -55,4 +66,99 @@ public static void dropTables() throws SQLException, ClassNotFoundException {
+  }
+
+  @Before(order = 2, value = "@POSTGRESQL_DATATYPES_TEST")
+  public static void createDatatypesTables() throws SQLException, ClassNotFoundException {
+    PostgresqlClient.createSourceDatatypesTable(PluginPropertyUtils.pluginProp("sourceTable"),
+                                       PluginPropertyUtils.pluginProp("schema"));
+    PostgresqlClient.createTargetDatatypesTable(PluginPropertyUtils.pluginProp("targetTable"),
+                                       PluginPropertyUtils.pluginProp("schema"));
+  }
+
+  @Before(order = 1, value = "@BQ_SINK_TEST")
+  public static void setTempTargetBQTableName() {
+    String bqTargetTableName = "E2E_TARGET_" + UUID.randomUUID().toString().replaceAll("-", "_");
+    PluginPropertyUtils.addPluginProp("bqTargetTable", bqTargetTableName);
+    BeforeActions.scenario.write("BQ Target table name - " + bqTargetTableName);
+  }
+
+  @After(order = 1, value = "@BQ_SINK_TEST")
+  public static void deleteTempTargetBQTable() throws IOException, InterruptedException {
+    String bqTargetTableName = PluginPropertyUtils.pluginProp("bqTargetTable");
+    try {
+      BigQueryClient.dropBqQuery(bqTargetTableName);
+      BeforeActions.scenario.write("BQ Target table - " + bqTargetTableName + " deleted successfully");
+      PluginPropertyUtils.removePluginProp("bqTargetTable");
+    } catch (BigQueryException e) {
+      if (e.getMessage().contains("Not found: Table")) {
+        BeforeActions.scenario.write("BQ Target Table " + bqTargetTableName + " does not exist");
+      } else {
+        Assert.fail(e.getMessage());
+      }
+    }
+  }
+
+  /**
+   * Create BigQuery table.
+   */
+  @Before(order = 1, value = "@BQ_SOURCE_TEST")
+  public static void createTempSourceBQTable() throws IOException, InterruptedException {
+    createSourceBQTableWithQueries(PluginPropertyUtils.pluginProp("CreateBQTableQueryFile"),
+                                   PluginPropertyUtils.pluginProp("InsertBQDataQueryFile"));
+  }
+
+  @After(order = 1, value = "@BQ_SOURCE_TEST")
+  public static void deleteTempSourceBQTable() throws IOException, InterruptedException {
+    String bqSourceTable = PluginPropertyUtils.pluginProp("bqSourceTable");
+    BigQueryClient.dropBqQuery(bqSourceTable);
+    BeforeActions.scenario.write("BQ source Table " + bqSourceTable + " deleted successfully");
+    PluginPropertyUtils.removePluginProp("bqSourceTable");
+  }
+
+  private static void createSourceBQTableWithQueries(String bqCreateTableQueryFile, String bqInsertDataQueryFile)
+    throws IOException, InterruptedException {
+    String bqSourceTable = "E2E_SOURCE_" + UUID.randomUUID().toString().substring(0, 5).replaceAll("-",
+                                                                                                   "_");
+
+    String createTableQuery = StringUtils.EMPTY;
+    try {
+      createTableQuery = new String(Files.readAllBytes(Paths.get(TestSetupHooks.class.getResource
+        ("/" + bqCreateTableQueryFile).toURI()))
```

Comment:
Where is ```bqCreateTableQueryFile``` which you are using to create the table.

Reply:
This is related to the creation with PostgreSQL sink plugin which is in progress, will push the sink changes once done.